content/reference/sql/version-control/dolt-sql-functions.md

```diff
@@ -656,3 +657,159 @@ returns just `D`. `DOLT_LOG('main...feature')` and `DOLT_LOG('feature...main')`
+
+## `DOLT_PATCH()`
+
+Generate the SQL statements needed to patch a table (or all tables) from a starting revision 
+to a target revision. This can be useful when you want to import data into Dolt from an external source, 
+compare differences, and generate the SQL statements needed to patch the original source. This command is
+equivalent of [`dolt diff -r sql` CLI command](../../cli.md#dolt-diff).
+Both schema and/or data diff statements are returned if applicable. Some data diff cannot be
+produced from incompatible schema changes; these are shown as warnings containing
+which table this occurred on.
+
+The order of the statements is that the schema patch comes first after the data patch. If patching all tables,
+then we recommend to turn off the foreign key checks before applying these patch statements in order of
+them returned to avoid conflicts.
+
+Getting SQL patch statements is only available as table function for now;
+the CLI `dolt patch` command will be supported in the future.
+
+### Privileges
+
+`DOLT_PATCH()` table function requires `SELECT` privilege for all tables if no table is defined or
+for the defined table only.
+
+### Options
+
+```sql
+DOLT_PATCH(<from_revision>, <to_revision>, <optional_tablename>)
+DOLT_PATCH(<from_revision..to_revision>, <optional_tablename>)
+DOLT_PATCH(<from_revision...to_revision>, <optional_tablename>)
+```
+
+The `DOLT_PATCH()` table function takes three arguments:
+
+- `from_revision` — the revision of the table data for the start of the patch. This argument is required. This may be a commit, tag, branch name, or other revision specifier (e.g. "main~", "WORKING", "STAGED").
+- `to_revision` — the revision of the table data for the end of the patch. This argument is required. This may be a commit, tag, branch name, or other revision specifier (e.g. "main~", "WORKING", "STAGED").
+- `from_revision..to_revision` — gets the two dot patch, or revision of table data between the `from_revision` and `to_revision`. This is equivalent to `dolt_patch(<from_revision>, <to_revision>, <tablename>)`.
+- `from_revision...to_revision` — gets the three dot patch, or revision of table data between the `from_revision` and `to_revision`, _starting at the last common commit_.
+- `tablename` — the name of the table containing the data and/or schema to patch. This argument is optional. When it's not defined, all tables with data and/or schema patch will be returned.
+
+### Schema
+
+```text
++------------------+--------+
+| field            | type   |
++------------------+--------+
+| statement_order  | BIGINT |
+| from_commit_hash | TEXT   |
+| to_commit_hash   | TEXT   |
+| table_name       | TEXT   |
+| diff_type        | TEXT   |
+| statement        | TEXT   |
++------------------+--------+
+```
+
+### Example
+
+```sql
+-- Get schema and data diffs on all table changes from head of main branch to head of feature branch
+CALL DOLT_PATCH('main', 'feature');
```

Comment:
Should this not be `select`, not `call`?

Reply:
good catch!